tests/test_views.py

```diff
@@ -708,3 +711,110 @@ def test_get_workflow_logs_unauthorized(app, default_user,
                          query_string={"user": random_user_uuid},
                          content_type='application/json')
         assert res.status_code == 403
+
+
+@pytest.mark.parametrize("state", [WorkflowStatus.created,
+                                   WorkflowStatus.failed,
+                                   WorkflowStatus.finished,
+                                   pytest.param(WorkflowStatus.deleted,
+                                                marks=pytest.mark.xfail),
+                                   pytest.param(WorkflowStatus.running,
+                                                marks=pytest.mark.xfail)])
+@pytest.mark.parametrize("hard_delete", [True, False])
+def test_delete_workflow(app,
+                         session,
+                         default_user,
+                         sample_yadage_workflow_in_db,
+                         state,
+                         hard_delete):
+    """Test deletion of a workflow in all possible states."""
+    sample_yadage_workflow_in_db.status = state
+    session.add(sample_yadage_workflow_in_db)
```

Comment:
@dinosk are you getting any advantage from having `sample_yadage_workflow_in_db` to be a yadage workflow? I was reading the fixture documentation and [it states that it is a serial workflow](https://github.com/reanahub/pytest-reana/blob/master/pytest_reana/fixtures.py#L435).. would it be ok if I change it to be serial? either way a fix in `pytest-reana` has to be applied.

Reply:
I used the yadage specification which was there so that would be the advantage. Why not add another one instead of changing this? I can imagine we would need both in the future. Thanks for pointing out the docstring, to be fixed. The fixture works as it is here, I saw that it was added in plugin.py